File: lab8/test/Programs/C/helloWorld.c
/****-*-c-*-***************************************************
* HelloWorld.c - a sample C program
*
*	Servesh Tiwari
*  01/26/2005
*
*	gcc 3.3.2 on Mandrake Linux 9.2
*
*	editor:	80 columns, tabstop=2
*
*	NOTE:  See instructions below main to compile
*
*	Revision History:
*		1/30/05	- fixed line comments, style, age logic	KS
*
***************************************************************/

/*
 	 These are block comments, over many lines!
	 Be careful, they do NOT nest.
   It is a good practice to comment code
*/

/* strictly speaking, there are no line comments in C */

#include <stdio.h>

int main()
{
  char name[20]; 
  int age = 0;

  printf( "Enter Your Name: " );
  scanf( "%s", name );
  printf( "Enter Your Age: " );
  scanf( "%d", &age );

	printf( "\n" );

  printf( "Hello World! Let me introduce %s!\n", name );
	printf( "Want to join us? We are off to " )

  if( age < 21 )
    printf( "the coffee house!\n" );
  else
    printf( "get some drinks!\n" );

	printf( "\n" );

  return 0;

}

/* 
   INSTRUCTIONS TO COMPILE AND RUN
   
   1. Use the editor of your choice to copy-paste the above lines into a file
      and save it as "helloWorld.c"
      I use emacs! So I'd be doing the following:
				 $emacs helloWorld.c & <RETURN> This would result in an emacs window
				 to open up. Select the text you want to copy, go back to the emacs
				 window, and press "ctrl + y". You should see the test pasted. Now
				 press "ctrl + x ctrl + s". This should save the file.

   2. To Compile type 
      [blah@localhos]$ gcc -o helloWorld helloworld.c
      and press enter. If you have gcc, and you pasted the file correctly,
      you should see a file helloWorld in your working directory (use ls to
      list the contents).  Note that it already has execute permissions set.
      
   3. To Execute:
      [blah@localhost]$ ./helloWorld <RETURN>

   4. Here is what it said when I tried to run it:
      [blah@localhost]$ ./helloWorld
      Enter Your Name: Blah
      Enter Your Age: 21

      Hello World! Let me introduce Blah!
      Want to join us? We are off to get some drinks!

      [blah@localhost]$
*/

File: lab8/test/Programs/Perl/helloWorld.pl
#!/usr/bin/perl -w
#
# helloWorld.pl - a sample Perl script
# 
# Servesh Tiwari
# 01/26/05
#
#	gcc 3.3.2 on Mandrake Linux 9.2
#
#	editor:	80 columns, tabstop=2
#
#	NOTE:  See instructions below main to run
#
#	NOTE:  the `-w' flag, above, just turns on warnings (good sanity checks)
#
#	Revision History:
#		1/30/05	- modified format, fixed age logic	KS
#
###########################################################

# this is a line comment
# perl does not have multiple line (block) commenting, so need a "#" for
# each line
# the first line is the location where the perl binary is in your system
# if you are using the department machine, you will need /usr/local/bin/perl

print "\nEnter Your Name: ";
$name=<STDIN>;
chomp $name;	# get rid of trailing newline, which was stored, above
print "Enter Your Age: ";
$age=<STDIN>;
chop $age;

printf "\nHello World! Let me introduce %s\n", $name;
printf "Want to join us? We are off to ";

if( $age < 21 )
{
    printf "the coffee house!\n";
}
else
{
    printf "get some drinks!\n";
}

printf "\n";


# INSTRUCTIONS IN RUNNING THE PERL SCRIPT

# 1. copy the above content to a file (say, helloWorld.pl)
# 2. Change the permissions of this file so that you can execute it.
#    You can use "chmod" command. e.g. "chmod 744 helloWorld.pl"
# 3. Type ./helloWorld.pl <RETURN>
# 4. If you get some error, it could be that your perl binary is not in the 
#    path specified in the first line. You will need to change that.
# 5. In case you were able to run it, this is what you are likely to see.
#    [blah@localhost]$ ./helloWorld.pl
#    Enter Your Name:blah
#    Enter Your Age:20
#    Hello World! Let me introduce blah
#    Want to join us? We are off to get some drinks!
#    [blah@localhost]$

